cmd/apps/istio_app.go

```diff
@@ -28,119 +30,136 @@ func MakeInstallIstio() *cobra.Command {
+	istio.Flags().String("memory", "100Mi", "Allocate Memory resource")
 
 	istio.Flags().StringArray("set", []string{},
 		"Use custom flags or override existing flags \n(example --set prometheus.enabled=false)")
 
-	istio.RunE = func(command *cobra.Command, args []string) error {
-		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
-		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
-			return err
+	istio.PreRunE = func(command *cobra.Command, args []string) error {
+		_, err := command.Flags().GetString("version")
+		if err != nil {
+			return fmt.Errorf("error with --version usage: %s", err)
 		}
-		wait, _ := command.Flags().GetBool("wait")
 
-		namespace, _ := command.Flags().GetString("namespace")
+		_, err = command.Flags().GetString("namespace")
+		if err != nil {
+			return fmt.Errorf("error with --namespace usage: %s", err)
+		}
 
-		if namespace != "istio-system" {
-			return fmt.Errorf(`to override the "istio-system" namespace, install Istio via helm manually`)
+		_, err = command.Flags().GetString("istio-namespace")
+		if err != nil {
+			return fmt.Errorf("error with --istio-namespace usage: %s", err)
 		}
 
-		arch := k8s.GetNodeArchitecture()
-		fmt.Printf("Node architecture: %q\n", arch)
+		_, err = command.Flags().GetString("profile")
+		if err != nil {
+			return fmt.Errorf("error with --profile usage: %s", err)
+		}
 
-		if arch != IntelArch {
-			return fmt.Errorf(OnlyIntelArch)
+		_, err = command.Flags().GetString("cpu")
+		if err != nil {
+			return fmt.Errorf("error with --cpu usage: %s", err)
 		}
 
-		userPath, err := config.InitUserDir()
+		_, err = command.Flags().GetString("memory")
 		if err != nil {
-			return err
+			return fmt.Errorf("error with --memory usage: %s", err)
 		}
 
-		clientArch, clientOS := env.GetClientArch()
+		_, err = command.Flags().GetString("kubeconfig")
+		if err != nil {
+			return fmt.Errorf("error with --kubeconfig usage: %s", err)
+		}
 
-		fmt.Printf("Client: %q, %q\n", clientArch, clientOS)
+		_, err = command.Flags().GetStringArray("set")
+		if err != nil {
+			return fmt.Errorf("error with --set usage: %s", err)
+		}
 
-		log.Printf("User dir established as: %s\n", userPath)
+		return nil
+	}
 
-		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+	istio.RunE = func(command *cobra.Command, args []string) error {
+		version, _ := command.Flags().GetString("version")
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		namespace, _ := command.Flags().GetString("namespace")
+		istioNamespace, _ := command.Flags().GetString("istio-namespace")
+		profile, _ := command.Flags().GetString("profile")
+		cpu, _ := command.Flags().GetString("cpu")
+		memory, _ := command.Flags().GetString("memory")
+		setOverrides, _ := command.Flags().GetStringArray("set")
 
-		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS)
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
```

Comment:
You need to block installation on armhf.

Reply:
Done